### DISCUSSION TITLE
Why is Maven POM its own language?

### DISCUSSION THREAD

**Discussion by Nixinova** [github-linguist/linguist#5189] [Q&A]
`languages.yml` lists Maven POM, filename pom.xml, as its own language even though it is entirely XML, as evidenced by its [docs](https://maven.apache.org/guides/introduction/introduction-to-the-pom.html):

> A Project Object Model or POM is the fundamental unit of work in Maven. It is an XML file that contains information about the project[...]

Should it be removed from `languages.yml`?

**Comment by Alhadis** [comment]
> Should it be removed from `languages.yml`?

No, because its `tm_scope` field indicates that it has a dedicated grammar for syntax highlighting:

https://github.com/github/linguist/blob/c461bc01486b15ca5460f6640e74ece32a32e0ce/lib/linguist/languages.yml#L3275-L3277

`SVG` and `XML Property List` are two important examples of XML-based formats that benefit from having specialised grammars: the SVG grammar highlights embedded JavaScript and CSS, for example, while XML property lists benefit from error highlighting due to their highly-specific and rigid structures.

That being said, I don't know how much of an improvement the Maven POM grammar (`text.xml.pom`) is over the normal XML grammar (`text.xml`). If the difference is negligible or noticeably worse, then yes, `Maven POM` should be rolled into `XML`.

**Reply by Nixinova** [reply]
At a glance the syntax highlighting seems identical and pom.xml doesn't seem to have any unique features.

**Reply by Nixinova** [reply]
Although, comparing the files, [XML](https://github.com/textmate/xml.tmbundle/blob/715363040b087a0c6720800e6dac30f9708b3f23/Syntaxes/XML.plist) is 573 lines while [Maven POM](https://github.com/textmate/maven.tmbundle/blob/180999aa43b2dbb3d3953f0196e766420ec34553/Syntaxes/Maven%20POM.tmLanguage) is 1226 lines so there must be something different, though the POM tmLanguage does seem to be [12 years old](https://github.com/textmate/maven.tmbundle/commits/master/Syntaxes/Maven%20POM.tmLanguage).

**Reply by Nixinova** [reply]
Should they be merged?

**Reply by Nixinova** [reply]
Looking at the file closely it just seems to tokenise each tag separately (which doesn't affect the syntax highlighting).

**Comment by Nixinova** [comment]
Either way, it should still be grouped under XML, as it is currently its own language.

**Reply by Alhadis** [reply]
I concur. This oversight probably fell through the cracks when Maven POM was originally added, as the pairing wouldn't have made a difference on XML's (lack of) influence on repository classification.

**Reply by Nixinova** [reply]
Opened #5236
